api/specs.proto

```diff
@@ -341,6 +341,11 @@ message NetworkSpec {
 	// swarm internally created only and it was identified by the name
 	// "ingress" and the label "com.docker.swarm.internal": "true".
 	bool ingress = 7;
+
+	// ConfigFrom indicates that the network specific configuration
+	// for this network will be provided via another network, locally
+	// on the node where this network is being plumbed.
+	string config_from = 8;
```

Comment:
When the user needs to create a cluster wide node-local (say macvlan) network, he needs to create on the nodes a configuration-only network which contains the node specific configurations (subnet, parent interface,...). He will then create the `--scope=swarm` level macvlan network specifying the config network which provides the node-local parameters with the `--config-from` flag.

When the creation of the dynamic, swarm network  is requested on the node, libnetwork will look for the configuration network to supply the node-local configurations.